Update full tutorials from develop (#567)
Avaiga/taipy-doc: docs/tutorials/complete_applications/complete_application/step_02/ReadMe.md

Reviewer: Please rename to step_02.md

> > You can download the full code [here](https://github.com/Avaiga/taipy-getting-started/tree/develop/src).

Reviewer: Same remark: link obsolete

# Algorithms used

> The application includes functions for data cleaning, baseline prediction, machine learning (ML) prediction, computing metrics, and creating a dataset for displaying predictions.

Reviewer 2: ```suggestion
The application contains functions for different tasks, such as cleaning data, making baseline predictions, using machine learning (ML) for predictions, calculating metrics, and generating a dataset to display the predictions.
```

> The application includes functions for data cleaning, baseline prediction, machine learning (ML) prediction, computing metrics, and creating a dataset for displaying predictions.

Reviewer: ```suggestion
The application includes functions for data cleaning, baseline prediction, machine
learning (ML) prediction, and creating a dataset for displaying output
predictions.
```

> `clean_data` is responsible for cleaning the initial dataset by converting the 'Date' column to datetime format. It takes an initial DataFrame as input and returns a cleaned copy of the DataFrame.

Reviewer 2: ```suggestion
The function `clean_data` handles the task of cleaning the initial dataset. It does this by converting the 'Date' column into datetime format. It takes an initial DataFrame as input and provides a cleaned copy of that DataFrame as output.
```

> `clean_data` is responsible for cleaning the initial dataset by converting the 'Date' column to datetime format. It takes an initial DataFrame as input and returns a cleaned copy of the DataFrame.

Reviewer: ```suggestion
## Clean data

`clean_data` is responsible for cleaning the initial dataset by converting the 'Date' column
to datetime format. It takes an initial DataFrame as input and returns a cleaned copy of the
DataFrame.
```

```python
def clean_data(initial_dataset: pd.DataFrame):
    print("     Cleaning data")
    initial_dataset['Date'] = pd.to_datetime(initial_dataset['Date'])
    cleaned_dataset = initial_dataset.copy()
    return cleaned_dataset
```

> `predict_baseline()` and `predict_ml()` predict values based on the `cleaned dataset`. It takes the cleaned DataFrame, the number of predictions to make (`n_predictions`), a specific date (`day`), and a maximum capacity value (`max_capacity`).

Reviewer: ```suggestion

## Predictions:

`predict_baseline()` and `predict_ml()` returns prediction values from the cleaned
DataFrame (`cleaned_dataset`), the number of predictions to make (`n_predictions`), a
specific date (`day`), and a maximum capacity value (`max_capacity`).
```

> They first select the training dataset up to the specified date. Then, it performs some computation or manipulation to give predictions. Predictions cannot exceed the maximum capacity.

Reviewer 2: ```suggestion
First, they pick the training dataset up to the date mentioned. After that, they perform certain calculations or adjustments to make predictions. These predictions must not surpass the maximum limit.
```

```python
def predict_baseline(cleaned_dataset: pd.DataFrame, n_predictions: int, day: dt.datetime, max_capacity: int):
    print("     Predicting baseline")
    train_dataset = cleaned_dataset[cleaned_dataset['Date'] < day]

    predictions = train_dataset['Value'][-n_predictions:].reset_index(drop=True)
    predictions = predictions.apply(lambda x: min(x, max_capacity))
    return predictions

# This is the function that will be used by the task
def predict_ml(cleaned_dataset: pd.DataFrame, n_predictions: int, day: dt.datetime, max_capacity: int):
    print("     Predicting with ML")
    # Select the train data
    train_dataset = cleaned_dataset[cleaned_dataset["Date"] < day]

    # Fit the AutoRegressive model
    model = AutoReg(train_dataset["Value"], lags=7).fit()

    # Get the n_predictions forecasts
    predictions = model.forecast(n_predictions).reset_index(drop=True)
    predictions = predictions.apply(lambda x: min(x, max_capacity))
    return predictions
```

> `create_predictions_dataset()` creates a predictions dataset for visualization purposes. It takes the predicted baseline values (`predictions_baseline`), ML predicted values (`predictions_ml`), a specific date (`day`), the number of predictions to make (`n_predictions`), and the cleaned dataset (`cleaned_data`).The function returns a DataFrame containing the date, historical values, ML predicted values, and baseline predicted values.

Reviewer: ```suggestion

## Output dataset

`create_predictions_dataset()` creates a predictions dataset for visualization purposes. It
takes the predicted baseline values (`predictions_baseline`), ML predicted values
(`predictions_ml`), a specific date (`day`), the number of predictions to make
(`n_predictions`), and the cleaned dataset (`cleaned_data`).The function returns a DataFrame
containing the date, historical values, ML predicted values, and baseline predicted values.
```

```python
def create_predictions_dataset(predictions_baseline, predictions_ml, day, n_predictions, cleaned_data):
    ....

    # Columns : [Date, Historical values, Predicted values (ml and baseline)]
    predictions_dataset = pd.concat([temp_df["Date"],
                                    historical_values,
                                    predicted_values_ml,
                                    predicted_values_baseline], axis=1)
    
    return predictions_dataset
```

> ## Entire code (algorithms/algorithms.py)

Reviewer: What do you think about showing the whole graph here? It is self explanatory.

![image](https://github.com/Avaiga/taipy-doc/assets/88036007/b0dd5ea2-8645-4f10-8c6a-d8aa232ace5c)

Of course we have to beautify it.

```suggestion
## Entire code (algorithms/algorithms.py)

Chaining all the functions together can be represented as a following graph:

```

    print("     Predicting baseline")
    # Select the train data
    train_dataset = cleaned_dataset[cleaned_dataset['Date'] < day]

    predictions = train_dataset['Value'][-n_predictions:].reset_index(drop=True)
    predictions = predictions.apply(lambda x: min(x, max_capacity))
    return predictions

# This is the function that will be used by the task
def predict_ml(cleaned_dataset: pd.DataFrame, n_predictions: int, day: dt.datetime, max_capacity: int):
    print("     Predicting with ML")
    # Select the train data
    train_dataset = cleaned_dataset[cleaned_dataset["Date"] < day]

    # Fit the AutoRegressive model
    model = AutoReg(train_dataset["Value"], lags=7).fit()

    # Get the n_predictions forecasts
    predictions = model.forecast(n_predictions).reset_index(drop=True)
    predictions = predictions.apply(lambda x: min(x, max_capacity))
    return predictions

def compute_metrics(historical_data, predicted_data):
    historical_to_compare = historical_data[-len(predicted_data):]['Value']
    rmse = mean_squared_error(historical_to_compare, predicted_data)
    mae = mean_absolute_error(historical_to_compare, predicted_data)
    return rmse, mae

> 

Reviewer: It is not described anywhere. Do we want to remove it?
